services/src/main/java/org/fao/geonet/services/metadata/Insert.java

```diff
@@ -40,75 +50,105 @@
 import org.fao.geonet.kernel.setting.SettingManager;
 import org.fao.geonet.repository.MetadataRepository;
 import org.fao.geonet.repository.Updater;
-import org.fao.geonet.services.NotInReadOnlyModeService;
 import org.fao.geonet.utils.Xml;
 import org.jdom.Element;
+import org.jdom.JDOMException;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseBody;
 
-import java.nio.file.Path;
-import java.util.ArrayList;
-import java.util.List;
-import java.util.Map;
-import java.util.UUID;
-import javax.annotation.Nonnull;
+import com.google.common.collect.Maps;
+
+import jeeves.constants.Jeeves;
+import jeeves.server.context.ServiceContext;
+import jeeves.server.sources.http.ServletPathFinder;
+import jeeves.services.ReadWriteController;
 
 /**
- * Inserts a new metadata to the system (data is validated).
+ * Insert a metadata into the catalog. You have to be logged in as editor to use
+ * this service.
+ * 
+ * <ul>
+ * <li>String <strong>data</strong>: XML string of the metadata</li>
+ * <li>Integer <strong>group</strong>: Id of the group to insert this metadata
+ * into</li>
+ * <li>[y|n]<strong>template</strong> (optional): Is it a template or a
+ * metadata?</li>
+ * <li>String <strong>styleSheet</strong> (optional): Stylesheet to apply to the
+ * inserted metadata. Based on xsl/conversion/import</li>
+ * <li>[on|off] <strong>validate</strong> (optional): Enable validation for this
+ * metadata</li>
+ * <li>String <strong>schema</strong> (optional): Schema of this metadata. If
+ * not defined, it will try to extract it from the metadata structure</li>
+ * <li>String <strong>category</strong> (optional): Category to assign to the
+ * metadata</li>
+ * <li>String <strong>extra</strong> (optional): Extra data</li>
+ * <li>[generateUuid] <strong>uuidAction</strong> (optional): Action to do with
+ * the uuid</li>
+ * </ul>
+ * 
  */
-public class Insert extends NotInReadOnlyModeService {
-    //--------------------------------------------------------------------------
-    //---
-    //--- Init
-    //---
-    //--------------------------------------------------------------------------
 
-    private Path stylePath;
-
-    public void init(Path appPath, ServiceConfig params) throws Exception {
-        this.stylePath = appPath.resolve(Geonet.Path.IMPORT_STYLESHEETS);
+@Controller("md.insert")
```

Comment:
I already removed them but I can rollback that change and still maintain the xml.metadata version of the service. I let you decide :smile: